### PR TITLE
Document command-line-interface of importers

### PR DESCRIPTION
**Pull request by ashmaroli**
A slimmer version of #503:
**Enhance just the dedicated page for importers.**

### CODE REVIEW

**Comment by ashmaroli** [comment]
Deploy Preview URL:
https://deploy-preview-507--jekyll-import.netlify.app/

File: docs/_importers/behance.md
Comment (parkr) [inline docs/_importers/behance.md:19]: The plugin you wrote is super cool and the final output is amazing! I'm kind of sad that a reader of this Jekyll site source wouldn't actually get to see much of the content that they'd view in the output. It simultaneously shows the power of Jekyll ("you can inject all sorts of content at build time!") and the drawback of that power ("a lot of the key content on the site isn't in the source files"). I wonder if it would make sense for either (1) us to generate the content into the site source which we push up to the repo or (2) the plugin to inject a variable into the Liquid template so our page can at least have a reference like `{{ auto_generated_usage_guide }}`. Some food for thought. Right now it's very magical but I'm debating if maybe it separates the source and final output too much?

File: docs/_importers/blogger.md
Comment (parkr) [inline docs/_importers/blogger.md:4]: Side note: could we lexically alphabetize the list of importers and set these in a plugin or in Liquid too? This seems a bit inefficient to list them all here. When adding a new importer, the contributor has to know the order of their new importer in the context of other importers which we could easily get a computer to figure that out.

File: docs/_layouts/importer.html
Comment (parkr) [inline docs/_layouts/importer.html:23]: [S9Y Database importer has a load of params (19, to be exact). This makes this slightly harder to see since there's so much horizontal scrolling.](https://deploy-preview-507--jekyll-import.netlify.app/docs/s9ydatabase/) Is it at all possible to make this a multiline output?

```text
jekyll import s9ydatabase \
  --dbname DB \
  --socket SOCKET \
...
```

Or do you think that would take up too much space? 🤔

File: docs/_plugins/importer_metadata.rb
Comment (parkr) [inline docs/_plugins/importer_metadata.rb:52]: Would be a good idea to add `o.default_value` or something to Mercenary, then we wouldn't need to rely on this. https://github.com/jekyll/mercenary/blob/master/lib/mercenary/option.rb

File: docs/_config.yml
Comment (ashmaroli) [reply docs/_config.yml:19]: Now that you mention it, the definition does look like *JSON*. But actually, it is just YAML.

But I shall change this to the suggested format for sure. :)

File: docs/_importers/behance.md
Comment (ashmaroli) [reply docs/_importers/behance.md:19]: For hosted metadata, we could generate another data file for CLI information using a Rake task just like we have one for the list of importer-dependencies.

But that is less exciting (for me, atleast) than having content generated automatically at runtime, especially because contributors / maintainers have to ensure the data file is updated as needed.
For example, say we add a new option to an importer or we make edits to the option-description, then
  - the data file has to be explicitly updated to ensure the change gets documented.
  - Maintainers would have to explicitly check if the pull request has the data file updated to match the proposed change before merging the PR.

Regarding the Liquid object `autogenerated...`, that too *is possible*, but I am not yet clear on the suggested use-case.

File: docs/_importers/blogger.md
Comment (ashmaroli) [reply docs/_importers/blogger.md:4]: Technically, this is not needed at all. At least with modern Jekyll because basically there's already the concept of `next_doc` and `prev_doc` for `Jekyll::Document` objects.
These were removed in the other pull request that made wholesale changes to the docs but left it untouched here to maintain the narrow scope of current PR.

File: docs/_layouts/importer.html
Comment (ashmaroli) [reply docs/_layouts/importer.html:23]: This block was just meant to quickly illustrate the CLI options. Technically for this importer, all of those options are *optional*. Therefore,
`jekyll import s9ydatabase` will work just as fine.
When making the block multiline, the end output is ugly (noisy) and we have to decide where exactly to draw the line to balance readability and accessibility. Besides, the table below lists all the options verbosely anyways.

File: docs/_plugins/importer_metadata.rb
Comment (ashmaroli) [reply docs/_plugins/importer_metadata.rb:52]: This is indeed a good idea. But is Mercenary still actively maintained?

File: docs/_importers/behance.md
Comment (parkr) [reply docs/_importers/behance.md:19]: Good points. Let's leave it how you have it now.

File: docs/_layouts/importer.html
Comment (parkr) [reply docs/_layouts/importer.html:23]: Ok makes sense. I thought they were all required when reading the example invocation. If there's a default value for the flag, maybe we can use the common convention of wrapping the argument in square brackets like `--socket [SOCKET]`?

File: docs/_plugins/importer_metadata.rb
Comment (parkr) [reply docs/_plugins/importer_metadata.rb:52]: It's in the Jekyll org so we can modify it to include this! It hasn't gotten love recently since it did everything we wanted, but we could take some time to improve it. 😄

**Review by parkr — APPROVED** [review]

File: docs/_layouts/importer.html
Comment (ashmaroli) [reply docs/_layouts/importer.html:23]: Modifying the long option here (i.e. bracketing the optional ARG) will cause a discrepancy between what's documented here and what gets generated in the terminal on invoking the CLI with `--help` switch.
Therefore, the best place to make this change is at the source-code level itself.

**Comment by ashmaroli** [comment]
@jekyllbot: merge +doc

**Pull request merged**
